### ISSUE TITLE
`no-unused-array-method-return`: entries() and keys() should also require array receiver (same rationale as #3434)

### ISSUE THREAD

**Issue by mahirhir** [sindresorhus/eslint-plugin-unicorn#3456]
PR #3434 added `values` to the internal `methodsRequiringArrayReceiver` set so that `Set.values()`, `Map.values()` etc. don't incorrectly trigger the rule. The same problem exists for `entries` and `keys`, which also exist on `Map` and `Set` (and other iterables) and are not array-specific:

```js
// should NOT be flagged — Map/Set, not arrays
const map = new Map([['a', 1]]);
map.entries(); // currently flagged incorrectly
map.keys();    // currently flagged incorrectly
```

Both share the exact same rationale as `values`: they are common methods on non-array iterables, so triggering without confirming an array receiver produces false positives.

Fix would be adding `'entries'` and `'keys'` to `methodsRequiringArrayReceiver` alongside `'values'`.

**Comment by sindresorhus** [comment]
I don't think this needs a rule change.

The concrete `new Map()` / `new Set()` cases are already not reported on current `main`.

The only similar case that still reports is an unknown receiver, for example `const map = getMap(); map.entries();`. That's intentional for this syntax-only rule. If we require a known array receiver for `entries` and `keys`, we would also stop catching real array cases like `const array = getArray(); array.entries();`.

This is also not quite the same as #3434. Drizzle's `.values({...})` is a non-iterator builder API, while `Map#entries`, `Set#entries`, and `Map#keys` all return iterators, so ignoring them is still dead code.
